packages/amplify-codegen/src/commands/models.js

```diff
@@ -73,9 +73,22 @@ async function generateModels(context) {
   const generateIndexRules = readFeatureFlag('codegen.generateIndexRules');
   const emitAuthProvider = readFeatureFlag('codegen.emitAuthProvider');
 
-  const enableDartNullSafety =
-    projectConfig.frontend === 'flutter' ? validateDartSDK(context, projectRoot) && readFeatureFlag('codegen.enableDartNullSafety') : false;
-
+  let enableDartNullSafety = readFeatureFlag('codegen.enableDartNullSafety');
+
+  if (projectConfig.frontend === 'flutter') {
+    const isMinimumDartVersionSatisfied = validateDartSDK(context, projectRoot);
+    context.print.warning(`Detected feature flag: “enabledartnullsafety : ${enableDartNullSafety}”`);
+    if (isMinimumDartVersionSatisfied && enableDartNullSafety) {
+      context.print.warning(
+        'Generating Dart Models with null safety. To opt out of null safe models, turn off the “enabledartnullsafety” feature flag.',
+      );
+    } else {
+      enableDartNullSafety = false;
```

Comment:
Do we need to set this to false?

Reply:
If the minimum version is false and ff is true, it will parse true to modelgen unless we change it to false.